### DISCUSSION TITLE
use 'code' property of lock entity

### DISCUSSION THREAD

**Discussion by MrJamsn** [ronengr/hass_nuki_bt#48] [Ideas]
the HA lock entity supports codes. https://developers.home-assistant.io/docs/core/entity/lock/
It would be nice to have this possibility to use it in order to avoid that everybody with access to HA can also use the entity to open the door. What do you think?

**Comment by ronengr** [comment]
from what i could tell, this is a way to pass a code to the lock for locks that require a code.
I don't see a support for verifying the code in HA itself, and I'm not sure that writing such implementation specifically for hass_nuki_bt is the correct approach as doing it in a secure way is not trivial.

A different approach to this issue is maybe using HA permissions
https://developers.home-assistant.io/docs/auth_permissions/

**Comment by MrJamsn** [comment]
ah okay :(

thanks, anyways - awesome piece of work!!
